File: 2.1/micro/c/relax.c
#include <stdio.h>

static inline double relax(double old, double du, double w) {
    double alt = du + w;
    return alt < old ? alt : old;
}

double step(double old, double du, double w) {
    return relax(old, du, w);
}

int main(void) {
    printf("%.2f\n", step(10.0, 3.0, 4.0));
    return 0;
}

File: 2.1/micro/rust/relax.rs
fn relax(old: f64, du: f64, w: f64) -> f64 {
    let alt = du + w;
    if alt < old {
        alt
    } else {
        old
    }
}

#[no_mangle]
pub fn step(old: f64, du: f64, w: f64) -> f64 {
    relax(old, du, w)
}

fn main() {
    println!("{:.2}", step(10.0, 3.0, 4.0));
}
